### DISCUSSION TITLE
Ubuntu can't see my hub bluetooth

### DISCUSSION THREAD

**Discussion by sgatu** [pybricks/support#1173] [Q&A]
Past days I've been working flawlessly connecting from my Ubuntu PC to the hub and writing commands through bluetooth, but suddenly I can't see the hub anymore in the bluetooth devices. At first I thought it was the hub, but I don't think is the problem because if I try to connect from my android phone I can see it, and even the same PC(dual boot) from Windows sees the hub. 

I tried to see if wireshark was seeing anything and nothing. Can't figure out what's happening. 

I tried flashing the firmware again, and still nothing, but if I flash the original lego firmware suddenly I can see it again, but that's not what I want, I want to work with pybricks firmware. I've also reinstalled my Ubuntu just to be sure, and I'm out of ideas...

**Comment by dlech** [comment]
Which firmware did you flash?

What does the light do on the hub when you have Pybricks firmware flashed and you power it on?

Are you using the same Ubuntu system to flash the firmware?

**Reply by sgatu** [reply]
Hi,

I've flashed both the last version on code.pybricks.com and beta.pybricks.com. The one I need is the beta one, but neither of them was getting detected by ubuntu bluetooth.

The light is flashing, it usually mean that is waiting for a bluetooth connection. When I go to my phone(Android) I can see the device, but ubuntu does not. Also I've just booted on Windows, because I have to get the work done, and I could connect to the hub and write programs to it. 

And yes I do, I've used same ubuntu to flash the firmware and that tries to connect to the hub. Even stranger I have a laptop also with Ubuntu and the laptop can see the hub. 

But the Ubuntu that does not see the hub can see other devices, like wireless headphones and such. It's like both don't want to communicate and can't figure out why.

**Reply by dlech** [reply]
What is the BlueZ version on the system that is not working? (`bluetoothctl --version`)

What kind of Bluetooth chip is it? (`lsusb`)

**Reply by sgatu** [reply]
I tried multiple versiones of BlueZ, now I'm using the one that came with Ubuntu 23.04, which is 5.66. 

The bluetooth chip is a wifi/bluetooth chip that came with the MB, **MediaTek RZ608** . The MB is a AORUS X570S ELITE AX. 

The thing is that it was working before, same system with Ubuntu, but suddenly stopped.

**Reply by dlech** [reply]
It seems strange that Wireshark would show literally nothing if flashing firmware works on the same system. I would expect it to at least show commands to put the adapter in scanning mode and possibly receiving advertisements from other devices if you have any.

**Reply by sgatu** [reply]
Yeah, it's pretty damn strange, I think it has something to do with my system but can't put my finger on it. I started working on Windows for now and everything is fine and I don't have the time to deal with it now, but if I find a solution I'll put it here. Thank you.
